File: algorithms/ConvertSortedArraytoBinarySearchTree/ConvertSortedArraytoBinarySearchTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    # @param {integer[]} nums
    # @return {TreeNode}
    def sortedArrayToBST(self, nums):
        if nums:
            mid = len(nums) / 2
            node = TreeNode(nums[mid])
            node.left = self.sortedArrayToBST(nums[:mid])
            node.right = self.sortedArrayToBST(nums[mid+1])
            return node

File: algorithms/ConvertSortedArraytoBinarySearchTree/ConvertSortedArraytoBinarySearchTree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
struct TreeNode* convert(int* nums, int low, int high);
struct TreeNode* sortedArrayToBST(int* nums, int numsSize) {
    return convert(nums, 0, numsSize - 1);
}

struct TreeNode* convert(int* nums, int low, int high) {
    if (low <= high) {
        int mid = (low + high) / 2;
        struct TreeNode* node = (struct TreeNode*)malloc(sizeof(struct TreeNode))

        node->val = nums[mid];
        node->left = conver(nums, low, mid-1);
        node->right = conver(nums, mid+1, hgih);

        return node;
    }
    return NULL;
}